include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -869,35 +907,18 @@ void
 FEPointEvaluation<n_components, dim, spacedim, Number>::reinit(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell,
   const ArrayView<const Point<dim>> &                         unit_points,
-  const dealii::internal::FEValuesImplementation::MappingRelatedData<dim,
-                                                                     spacedim>
+  const std::shared_ptr<internal::FEPointEvaluation::MappingData<dim, spacedim>>
     &precomputed_mapping_data)
 {
-  this->unit_points.resize(unit_points.size());
-  std::copy(unit_points.begin(), unit_points.end(), this->unit_points.begin());
-
   if (!poly.empty())
     {
-      // Check the mapping data for consistency.
-      if (update_flags_mapping & update_jacobians)
-        Assert(precomputed_mapping_data.jacobians.size() == unit_points.size(),
-               ExcDimensionMismatch(precomputed_mapping_data.jacobians.size(),
-                                    unit_points.size()));
-      if (update_flags_mapping & update_inverse_jacobians)
-        Assert(precomputed_mapping_data.inverse_jacobians.size() ==
-                 unit_points.size(),
-               ExcDimensionMismatch(
-                 precomputed_mapping_data.inverse_jacobians.size(),
-                 unit_points.size()));
-
-      if (update_flags_mapping & update_quadrature_points)
-        Assert(precomputed_mapping_data.inverse_jacobians.size() ==
-                 unit_points.size(),
-               ExcDimensionMismatch(
-                 precomputed_mapping_data.quadrature_points.size(),
-                 unit_points.size()));
-
       mapping_data = precomputed_mapping_data;
+
+      // As we got the data from the other field, we will typically not need
+      // to perform any operation, unless the user provided us with a
+      // `precomputed_mapping_data` that is not initialized to the present
+      // context.
+      mapping_data->compute(*mapping, cell, unit_points, update_flags_mapping);
```

Comment:
OK. The flags might be different between FPEs that share the same mapping (additional fields are filled). But if `cell` and `unit_points` is different, this looks like a bug in the code. Also, does that mean that we are modifying the internal state of the other FPE that has called this function before? We don't have any way to signal that its that has been invalidated.